### PR TITLE
Fix for unhandled exceptions in the Triggerer tests

### PR DESCRIPTION
**Pull request by drajguru**
Fix for unhandled exceptions raised in the triggerer tests on the queue listener thread.

The queue listener was being killed due to missing a reference to the triggerer_job. Fixing this resulted in console logging during tests. So changes also include disabling the console logging handler / queue listener during tests.

Also fixed a bug in tests/jobs/test_triggerer_job.py::test_trigger_create_race_condition_18392 which caused another unhandled exception.

----

```console
tests/jobs/test_triggerer_job.py::test_trigger_logging_sensitive_info                                                                                                               
  /usr/local/lib/python3.8/site-packages/_pytest/threadexception.py:73: PytestUnhandledThreadExceptionWarning: Exception in thread Thread-3                                         
                                                                                                                                                                                    
  Traceback (most recent call last):                                                                                                                                                
    File "/usr/local/lib/python3.8/threading.py", line 932, in _bootstrap_inner                                                                                                     
      self.run()                                                                                                                                                                    
    File "/usr/local/lib/python3.8/threading.py", line 870, in run                                                                                                                  
      self._target(*self._args, **self._kwargs)                                                                                                                                     
    File "/usr/local/lib/python3.8/logging/handlers.py", line 1487, in _monitor                                                                                                     
      self.handle(record)                                                                                                                                                           
    File "/usr/local/lib/python3.8/logging/handlers.py", line 1468, in handle                                                                                                       
      handler.handle(record)                                                                                                                                                        
    File "/opt/airflow/airflow/utils/log/trigger_handler.py", line 107, in handle                                                                                                   
      self.emit(record)                                                                                                                                                             
    File "/opt/airflow/airflow/utils/log/trigger_handler.py", line 96, in emit                                                                                                      
      h = self._get_or_create_handler(record.trigger_id, record.task_instance)                                                                                                      
    File "/opt/airflow/airflow/utils/log/trigger_handler.py", line 92, in _get_or_create_handler                                                                                    
      self.handlers[trigger_id] = self._make_handler(ti)                                                                                                                            
    File "/opt/airflow/airflow/utils/log/trigger_handler.py", line 87, in _make_handler                                                                                             
      h.set_context(ti=ti)                                                                                                                                                          
    File "/opt/airflow/airflow/utils/log/file_task_handler.py", line 243, in set_context                                                                                            
      local_loc = self._init_file(ti, identifier=identifier)                                                                                                                        
    File "/opt/airflow/airflow/utils/log/file_task_handler.py", line 533, in _init_file                                                                                             
      full_path = self.add_triggerer_suffix(full_path=full_path, job_id=ti.triggerer_job.id)                                                                                        
  AttributeError: 'NoneType' object has no attribute 'id'                                                                                                                           
                                                                                                                                                                                    
    warnings.warn(pytest.PytestUnhandledThreadExceptionWarning(msg))                                                                                                                
                                                                                                                                                                                    
tests/jobs/test_triggerer_job.py::test_trigger_lifecycle                                                                                                                            
  /usr/local/lib/python3.8/site-packages/_pytest/threadexception.py:73: PytestUnhandledThreadExceptionWarning: Exception in thread Thread-11                                        
                                                                                                                                                                                    
  Traceback (most recent call last):                                                                                                                                                
    File "/usr/local/lib/python3.8/threading.py", line 932, in _bootstrap_inner                                                                                                     
      self.run()                                                                                                                                                                    
    File "/usr/local/lib/python3.8/threading.py", line 870, in run
      self._target(*self._args, **self._kwargs)
    File "/usr/local/lib/python3.8/logging/handlers.py", line 1487, in _monitor
      self.handle(record)
    File "/usr/local/lib/python3.8/logging/handlers.py", line 1468, in handle
      handler.handle(record)
    File "/opt/airflow/airflow/utils/log/trigger_handler.py", line 107, in handle
      self.emit(record)
    File "/opt/airflow/airflow/utils/log/trigger_handler.py", line 96, in emit
      h = self._get_or_create_handler(record.trigger_id, record.task_instance)
    File "/opt/airflow/airflow/utils/log/trigger_handler.py", line 92, in _get_or_create_handler
      self.handlers[trigger_id] = self._make_handler(ti)
    File "/opt/airflow/airflow/utils/log/trigger_handler.py", line 87, in _make_handler
      h.set_context(ti=ti)
    File "/opt/airflow/airflow/utils/log/file_task_handler.py", line 243, in set_context
      local_loc = self._init_file(ti, identifier=identifier)
    File "/opt/airflow/airflow/utils/log/file_task_handler.py", line 533, in _init_file
      full_path = self.add_triggerer_suffix(full_path=full_path, job_id=ti.triggerer_job.id)
  AttributeError: 'NoneType' object has no attribute 'id'


tests/jobs/test_triggerer_job.py::test_trigger_create_race_condition_18392
  /usr/local/lib/python3.8/site-packages/_pytest/threadexception.py:73: PytestUnhandledThreadExceptionWarning: Exception in thread Thread-19
   
  Traceback (most recent call last):
    File "/usr/local/lib/python3.8/threading.py", line 932, in _bootstrap_inner
      self.run()
    File "/usr/local/lib/python3.8/threading.py", line 870, in run
      self._target(*self._args, **self._kwargs)
    File "/opt/airflow/airflow/jobs/triggerer_job_runner.py", line 341, in _execute
      self._run_trigger_loop()
    File "/opt/airflow/airflow/jobs/triggerer_job_runner.py", line 364, in _run_trigger_loop
      self.load_triggers()
    File "/opt/airflow/tests/jobs/test_triggerer_job.py", line 376, in load_triggers
      self.wait_for_runner_loop(runner_loop_count=2)
    File "/opt/airflow/tests/jobs/test_triggerer_job.py", line 367, in wait_for_runner_loop
      pytest.fail("did not observe 2 loops in the runner thread")
    File "/usr/local/lib/python3.8/site-packages/_pytest/outcomes.py", line 198, in fail
      raise Failed(msg=reason, pytrace=pytrace)
  Failed: did not observe 2 loops in the runner thread
   
    warnings.warn(pytest.PytestUnhandledThreadExceptionWarning(msg))
                                                                                                              
                                                                                                                                                                         
tests/jobs/test_triggerer_job.py::test_trigger_firing                                                                                                                               
  /usr/local/lib/python3.8/site-packages/_pytest/threadexception.py:73: PytestUnhandledThreadExceptionWarning: Exception in thread Thread-28                                        
                                                                                                                                                                                    
  Traceback (most recent call last):                                                                                                                                                
    File "/usr/local/lib/python3.8/threading.py", line 932, in _bootstrap_inner                                                                                                     
      self.run()                                                                                                                                                                    
    File "/usr/local/lib/python3.8/threading.py", line 870, in run                                                                                                                  
      self._target(*self._args, **self._kwargs)                                                                                                                                     
    File "/usr/local/lib/python3.8/logging/handlers.py", line 1487, in _monitor                                                                                                     
      self.handle(record)                                                                                                                                                           
    File "/usr/local/lib/python3.8/logging/handlers.py", line 1468, in handle                                                                                                       
      handler.handle(record)                                                                                                                                                        
    File "/opt/airflow/airflow/utils/log/trigger_handler.py", line 107, in handle                                                                                                   
      self.emit(record)                                                                                                                                                             
    File "/opt/airflow/airflow/utils/log/trigger_handler.py", line 96, in emit                                                                                                      
      h = self._get_or_create_handler(record.trigger_id, record.task_instance)                                                                                                      
    File "/opt/airflow/airflow/utils/log/trigger_handler.py", line 92, in _get_or_create_handler                                                                                    
      self.handlers[trigger_id] = self._make_handler(ti)                                                                                                                            
    File "/opt/airflow/airflow/utils/log/trigger_handler.py", line 87, in _make_handler                                                                                             
      h.set_context(ti=ti)                                                                                                                                                          
    File "/opt/airflow/airflow/utils/log/file_task_handler.py", line 243, in set_context                                                                                            
      local_loc = self._init_file(ti, identifier=identifier)
    File "/opt/airflow/airflow/utils/log/file_task_handler.py", line 533, in _init_file
      full_path = self.add_triggerer_suffix(full_path=full_path, job_id=ti.triggerer_job.id)
  AttributeError: 'NoneType' object has no attribute 'id'
   
    warnings.warn(pytest.PytestUnhandledThreadExceptionWarning(msg))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```

### CODE REVIEW

File: tests/jobs/test_triggerer_job.py [hunk 1]
```diff
@@ -51,6 +52,24 @@
 pytestmark = pytest.mark.db_test
 
 
+@pytest.fixture(autouse=True)
```
Comment (uranusjr) [inline tests/jobs/test_triggerer_job.py:55]:
```suggestion
@pytest.fixture(scope="module", autouse=True)
```

It doesn’t feel right to do this for _all_ tests. Either this or we want to manually apply this to tests that matter instead.

File: tests/jobs/test_triggerer_job.py [hunk 2]
```diff
@@ -51,6 +52,24 @@
...
+def disable_console_handler():
+    """
+    Prevent console handler from getting bound to the trigger_job_runner when setup_queue_listener is called during tests
+    """
+
+    class QueueListener_(logging.handlers.QueueListener):
```
Comment (uranusjr) [inline tests/jobs/test_triggerer_job.py]: Does this class need to be defined in the function scope?

Comment (drajguru) [reply tests/jobs/test_triggerer_job.py]: True, I just moved it out

File: tests/jobs/test_triggerer_job.py [hunk 3]
```diff
@@ -51,6 +52,24 @@
 pytestmark = pytest.mark.db_test
 
 
+class QueueListener_(logging.handlers.QueueListener):
+    def __init__(self, queue, *handlers, respect_handler_level):
+        no_console_handlers = []
+        for h in handlers:
+            if h.name != "console":
+                no_console_handlers.append(h)
```
Comment (uranusjr) [inline tests/jobs/test_triggerer_job.py]:
```suggestion
        no_console_handlers = (h for h in handlers if h.name != "console")
```

File: tests/models/test_trigger.py [hunk 4]
```diff
@@ -142,10 +142,12 @@ def test_submit_failure(session, create_task_instance):
     assert updated_task_instance.next_method == "__fail__"
 
 
-def test_assign_unassigned(session, create_task_instance):
+def test_assign_unassigned(session, create_task_instance, monkeypatch):
     """
     Tests that unassigned triggers of all appropriate states are assigned.
     """
+    # disable QueueListener to prevent logging during tests
+    monkeypatch.setattr("airflow.jobs.triggerer_job_runner.DISABLE_LISTENER", True)
```
Comment (uranusjr) [inline tests/models/test_trigger.py:150]: I wonder if there’s a better way… perhaps adding an argument on `TriggererJobRunner` for this? Not entirely sure.

File: tests/jobs/test_triggerer_job.py [hunk 5]
```diff
@@ -398,8 +427,11 @@ def handle_events(self):
     session.commit()
 
     job = Job()
+    job.prepare_for_execution()
     job_runner = TriggererJob_(job)
     job_runner.trigger_runner = TriggerRunner_()
+    # set job_id so that it can be used by FileTaskHandler.add_triggerer_suffix
+    job_runner.trigger_runner.job_id = job.id
```
Comment (uranusjr) [inline tests/jobs/test_triggerer_job.py]: Why is this relevant?

File: tests/jobs/test_triggerer_job.py [hunk 6]
```diff
@@ -361,7 +390,7 @@ class TriggererJob_(TriggererJobRunner):
         def wait_for_runner_loop(self, runner_loop_count):
             for _ in range(30):
                 time.sleep(0.1)
-                if getattr(self.trigger_runner, "call_count", 0) >= runner_loop_count:
+                if getattr(self.trigger_runner, "loop_count", 0) >= runner_loop_count:
```
Comment (uranusjr) [inline tests/jobs/test_triggerer_job.py:386]: Is there a way to detect this if the attribute gets renamed?

**Pull request closed without merging**